diff --git a/node_modules/@walletconnect/web3-provider/dist/cjs/index.js b/node_modules/@walletconnect/web3-provider/dist/cjs/index.js
index 31c8049..fd50a61 100644
--- a/node_modules/@walletconnect/web3-provider/dist/cjs/index.js
+++ b/node_modules/@walletconnect/web3-provider/dist/cjs/index.js
@@ -43,6 +43,7 @@ class WalletConnectProvider extends ProviderEngine {
             return this.send(payload);
         });
         this.send = (payload, callback) => tslib_1.__awaiter(this, void 0, void 0, function* () {
+            var _a;
             if (typeof payload === "string") {
                 const method = payload;
                 let params = callback;
@@ -59,7 +60,13 @@ class WalletConnectProvider extends ProviderEngine {
                 this.sendAsync(payload, callback);
                 return;
             }
-            return this.sendAsyncPromise(payload.method, payload.params);
+            if (payload.method === "eth_signTypedData_v4" && ((_a = this.walletMeta) === null || _a === void 0 ? void 0 : _a.name) === "MetaMask") {
+                const { result } = yield this.handleOtherRequests(payload);
+                return result;
+            }
+            else {
+                return this.sendAsyncPromise(payload.method, payload.params);
+            }
         });
         this.onConnect = (callback) => {
             this.connectCallbacks.push(callback);
